cloud-template/src/main/java/com/sequenceiq/cloudbreak/cloud/template/AbstractResourceConnector.java

```diff
@@ -116,8 +120,12 @@
         //context
         ResourceBuilderContext context = contextBuilders.get(platform).contextInit(cloudContext, auth, stack.getNetwork(), cloudResources, false);
 
+        //loadBalancer
+        List<CloudResourceStatus> cloudResourceStatuses = loadBalancerResourceService.deleteResources(context, auth, cloudResources, false);
+
         //compute
-        List<CloudResourceStatus> cloudResourceStatuses = computeResourceService.deleteResources(context, auth, cloudResources, false);
+        List<CloudResourceStatus> computeStatuses = computeResourceService.deleteResources(context, auth, cloudResources, false);
```

Comment:
What is the reason behind introducing a new group of resources?

Loadbalancers could be created with compute resource builders and the resource builder order could give the possibility to delete/create them at the right phase of resource building.

Reply:
loadBalancers are created after the rest of the stack with the ResourceConnector.launchLoadBalancer. We don't want to accidentally recreate compute resources here. 
They could be ordered with compute resource builders. but they'd always come afterwards, so the fact that it happens in their own flow step is fine.


additionally, when building with a compute resource builder, it's tied to a CloudInstance which doesn't exist for loadBalancers, and we wouldn't want it to. Additionally building it for a specific Cloudloadbalancer lets us handle adding an additional one.

Reply:
With Native AWS approach we did almost the same approach, used the `launchLoadBalancer` to create resources and created the compute resource builders for eliminating the load balancer related resources.
[AwsNativeLoadBalancerResourceBuilder](https://github.com/hortonworks/cloudbreak/blob/965345ea651fde41481d0cb1fb44fe7f84a3573c/cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/loadbalancer/AwsNativeLoadBalancerResourceBuilder.java#L28)

If we introduce a new phase of builders for load balancers, then I would suggest to create and delete resources within the same resource builder and not using the `ResourceConnector.launchLoadBalancer` for resource creation **OR** extend the dedicated `ResourceConnector` interface with the deletion of load balancer resources and wiring it to the right flow step and in this case the modification of the resource builder engine is also avoidable. cc: @lacikaaa @doktoric @hreeve-cloudera 
On the other hand I would not touch the resource builder's logic if it is not necessary just for deleting resources that have been created by not a builder.

Reply:
independent of where it's being used in the flow, having all the methods for creation and deletion of a resource in a single builder is useful for keeping a consistent image of how they are interacted with. 

Whether or not there needs to be a new flow step for the deletion of loadbalancers is probably outside the scope of this change.

Reply:
Yeah, it could be done this way, but this change is more risky as other cloud providers are also using the same logic in AbstractResourceConnector.java. So please create more unit tests for the new load balancer related builder modification in the 'common' logic and also make sure that clusters with and without load balancers could be created on AWS native variant for example.